workers/unity/Packages/com.improbable.gdk.core/Worker/ConnectionHandlers/WorldCommandMetaDataStorage.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections.Generic;
+using Improbable.Gdk.Core.Commands;
+
+namespace Improbable.Gdk.Core
+{
+    public class WorldCommandMetaDataStorage : ICommandMetaDataStorage
+        , ICommandPayloadStorage<WorldCommands.CreateEntity.Request>
+        , ICommandPayloadStorage<WorldCommands.DeleteEntity.Request>
+        , ICommandPayloadStorage<WorldCommands.ReserveEntityIds.Request>
+        , ICommandPayloadStorage<WorldCommands.EntityQuery.Request>
+    {
+        private readonly Dictionary<long, CommandContext<WorldCommands.CreateEntity.Request>> idToCreateEntityRequest =
+            new Dictionary<long, CommandContext<WorldCommands.CreateEntity.Request>>();
+
+        private readonly Dictionary<long, CommandContext<WorldCommands.DeleteEntity.Request>> idToDeleteEntityRequest =
+            new Dictionary<long, CommandContext<WorldCommands.DeleteEntity.Request>>();
+
+        private readonly Dictionary<long, CommandContext<WorldCommands.ReserveEntityIds.Request>>
+            idToReserveEntityIdsRequest =
+                new Dictionary<long, CommandContext<WorldCommands.ReserveEntityIds.Request>>();
+
+        private readonly Dictionary<long, CommandContext<WorldCommands.EntityQuery.Request>> idToEntityQueryRequest =
+            new Dictionary<long, CommandContext<WorldCommands.EntityQuery.Request>>();
+
+        private readonly Dictionary<uint, long> internalRequestIdToRequestId = new Dictionary<uint, long>();
+
+        public uint GetComponentId()
+        {
+            return 0;
+        }
+
+        public uint GetCommandId()
+        {
+            return 0;
+        }
+
+        public void RemoveMetaData(uint internalRequestId)
+        {
+            var requestId = internalRequestIdToRequestId[internalRequestId];
+            internalRequestIdToRequestId.Remove(internalRequestId);
+            idToCreateEntityRequest.Remove(requestId);
+            idToDeleteEntityRequest.Remove(requestId);
+            idToReserveEntityIdsRequest.Remove(requestId);
+            idToEntityQueryRequest.Remove(requestId);
```

Comment:
Can we optimise by checking the result of the `Remove` operations?

Reply:
Note to self - got to here

Reply:
Yes. Will OR them all.